spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -502,13 +500,18 @@ def __init__(self, parent=None):
 
         # Tab key behavior
         self.tab_indents = None
-        self.tab_mode = True # see CodeEditor.set_tab_mode
+        self.tab_mode = True  # see CodeEditor.set_tab_mode
 
         # Intelligent backspace mode
         self.intelligent_backspace = True
 
         # Automatic (on the fly) completions
-        self.automatic_completions = True
+        self.automatic_completions = \
+            CONF.get('editor', 'automatic_completions')
+        self.automatic_completions_after_chars = \
+            CONF.get('editor', 'automatic_completions_after_chars')
+        self.automatic_completions_after_ms = \
+            CONF.get('editor', 'automatic_completions_after_ms')

```

Comment:
This changes shouldn't be necessary because:

1. Options from our config system are applied to each CodeEditor at startup and also when an option is changed in our Preferences. If you that that's not the case, then it's a bug and we need to fix it.
2. There's no need for widgets to access our config system directly. Plugins (i.e. the ones that contain widgets) are the ones in charge of that.

Reply:
I can confirm that without this change, the config options are not being propagated here at startup, but *are* when the options are changed.

It seems like checking the `CONF` value here would solve the startup case, but I'm not sure if there's another way of fixing that. I can create an issue to track that and revert it here.

Reply:
https://github.com/spyder-ide/spyder/issues/10331